### PR TITLE
Neutral colors for progress bars and translation unit states

### PR DESCRIPTION
**Pull request by m-aciek**
## Proposed changes

Make progress bars use neutral colors (primary and neutral instead of info and danger). Updated also legend and unit states colors. Introduced new class for unit states, `.unit-state-bad` to differ units with problems from `.unit-state-todo`.

For me personally currently high amount of "danger" color for in progress translations causes an "UI anxiety".

## Checklist

- [x] Lint and unit tests pass locally with my changes.
- ~~[ ] I have added tests that prove my fix is effective or that my feature works.~~
- ~~[ ] I have added documentation to describe my feature.~~
- [x] I have squashed my commits into logic units.
- [x] I have described the changes in the commit messages.

## Other information

<details>
  <summary>Light before</summary>
  
hosted weblate org_projects_accrescent_ (1)

</details>

<details>
  <summary>Light after</summary>
  
hosted weblate org_projects_accrescent_

</details>
<details>
  <summary>Dark before</summary>
  
hosted weblate org_projects_accrescent_ (3)

</details>
<details>
  <summary>Dark after</summary>
  
hosted weblate org_projects_accrescent_ (2)
</details>

Legend in translation view:

<details>
  <summary>Light before</summary>
  
hosted weblate org_projects_accrescent_client_pl_ (2)

</details>

<details>
  <summary>Light after</summary>
  
hosted weblate org_projects_accrescent_client_pl_ (3)

</details>
<details>
  <summary>Dark before</summary>
  
hosted weblate org_projects_accrescent_client_pl_ (1)

</details>
<details>
  <summary>Dark after</summary>
  
hosted weblate org_projects_accrescent_client_pl_
</details>

Units view:

<details>
  <summary>Light before</summary>
  
hosted weblate org_browse_antennapod_website-general_fr__sort_by=labels offset=4 (3)

</details>
<details>
  <summary>Light after</summary>
  
hosted weblate org_browse_antennapod_website-general_fr__sort_by=labels offset=4 (5)

</details>
<details>
  <summary>Dark before</summary>
  
hosted weblate org_browse_antennapod_website-general_fr__sort_by=labels offset=4

</details>
<details>
  <summary>Dark after</summary>
  
hosted weblate org_browse_antennapod_website-general_fr__sort_by=labels offset=4 (4)

</details>

### CODE REVIEW

**Comment by nijel** [comment]
Let's discuss this change for the 5.1 release. In case it is found reasonable, additional changes need to be done as the same colors are used on translation details.

**Comment by nijel** [comment]
Also, these colors match colors on the strings listing. Using neutral colors here will make the status hardly visible.

**Comment by m-aciek** [comment]
I've pushed updating the legend in the translation view.

<details>
  <summary>Light before</summary>
  
hosted weblate org_projects_accrescent_client_pl_ (2)

</details>

<details>
  <summary>Light after</summary>
  
hosted weblate org_projects_accrescent_client_pl_ (3)

</details>
<details>
  <summary>Dark before</summary>
  
hosted weblate org_projects_accrescent_client_pl_ (1)

</details>
<details>
  <summary>Dark after</summary>
  
hosted weblate org_projects_accrescent_client_pl_
</details>

I think that using _danger_ color for untranslated strings incorrectly suggests that there's an issue with those strings. In "average" scenario of correctly translating a project I would expect a "gray" progress bar being replaced with "green == success" fragments until it becomes green entirely.

I'm not sure if the contrast of neutral color in legend is too low. I think making it darker requires Bootstrap config customization.

**Comment by m-aciek** [comment]
I've pushed changes to update the units view.

<details>
  <summary>Light before</summary>
  
hosted weblate org_browse_antennapod_website-general_fr__sort_by=labels offset=4 (3)

</details>
<details>
  <summary>Light after</summary>
  
hosted weblate org_browse_antennapod_website-general_fr__sort_by=labels offset=4 (2)

</details>
<details>
  <summary>Dark before</summary>
  
hosted weblate org_browse_antennapod_website-general_fr__sort_by=labels offset=4

</details>
<details>
  <summary>Dark after</summary>
  
hosted weblate org_browse_antennapod_website-general_fr__sort_by=labels offset=4 (1)

</details>

The primary color for approved looks very dark here, I wonder if keeping the info color here wouldn't be better.

File: weblate/templates/translation.html [hunk 1]
```diff
@@ -175,11 +175,13 @@ <h4 class="panel-title">
 <tr data-href="{{ translate_url }}?q={{ query }}" class="clickable-row">
 <th class="number">{{ count|intcomma }}</th>
 <td class="legend">
+    {% if name != "all" %}
     <div class="progress">
       {% if color %}
-        <div class="progress-bar progress-bar-{{ color }}" role="progressbar" aria-valuenow="100" aria-valuemin="0" aria-valuemax="100" style="width: 100%;"></div>
+        <div class="progress-bar{% if color != "primary" %} progress-bar-{{ color }}{% endif %}" role="progressbar" aria-valuenow="100" aria-valuemin="0" aria-valuemax="100" style="width: 100%;"></div>
```
[the remark this answers is not in the document]
Comment (m-aciek) [reply weblate/templates/translation.html:181]: It's just a legend color for a row, which is described in `full-cell` next to it. Potentially we could use `aria-labelled-by` with this `td`, but that would require introducing unique `id` on `full-cell` elements.

**Comment by nijel** [comment]
Sorry, we unfortunately didn't find to discuss this in time for the 5.1 release, postponing the  decision to 5.2.

File: weblate/templates/snippets/progress.html [hunk 2]
```diff
@@ -2,12 +2,9 @@
 
 <div class="progress">
```
Comment (nijel) [inline weblate/templates/snippets/progress.html:3]: Include the title from the last part of the progress bar which got removed:

```suggestion
<div class="progress" title="{% trans "Needs attention" %}">
```

Alternatively, keep the last part just to show the tooltip.

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Comment by PeterDurrer** [comment]
Hello @m-aciek , @nijel i have Update Weblate from very old Version to the New Version and Now the progress bar is with this neutral colors. Is there a way that i can configure that is used the old colors or that i can configure self the colors? For the Navigation this can be configure in weblate directly how can i do this for the progress? In our Company it was nice to see with colors in progess bar on what projects have open failed checks and what strings are need to control.

**Comment by nijel** [comment]
No, there is no such configuration option.

**Comment by PeterDurrer** [comment]
> No, there is no such configuration option.

Is there planned that this colors can be configured in Future? Or How can i create an Issue to implement a configure?

**Pull request merged**